src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -177,13 +186,26 @@ class JoinClauseSegment(ansi.JoinClauseSegment):
             ),
             Conditional(Dedent, indented_using_on=True),
         ),
-        # Note NATURAL joins do not support Join conditions
-        Sequence(
-            Ref("JoinKeywordsGrammar"),
```

Comment:
Just to confirm, did you intend to delete this block?

Reply:
Yes, ClickHouse does not support `NATURAL JOIN` and this block was added here by mistake